src/game/Unit.cpp

```diff
@@ -339,7 +339,7 @@ bool Unit::UpdateMeleeAttackingState()
     if (!victim || IsNonMeleeSpellCasted(false))
         return false;
 
-    if (!isAttackReady(BASE_ATTACK) && !(isAttackReady(OFF_ATTACK) && haveOffhandWeapon()))
+    if (!isAttackReady(BASE_ATTACK) && (haveOffhandWeapon() && !(isAttackReady(OFF_ATTACK))))
```

Comment:
Actually, the outcome of the two conditions is exactly the same, mine is just cleaner as it clearly shows what it does, where the initial one is a bit obscure.

It basically just went from a NAND to a more transparent AND NOT function instead.